### ISSUE TITLE
LibJS: `let` variables are not captured by value inside a for loop

### ISSUE THREAD

**Issue by tcl3** [LadybirdBrowser/ladybird#602]
When a `let` variable is captured by a function, it's value at the time it was captured should be used. Currently, Ladybird behaves as if a `var` variable was being captured.

Example:

```html
<!DOCTYPE html>
<script>
    let result = "";
    let functionList = [];
    for (let i = 0; i < 9; i++) {
      functionList.push(function() {
          return i;
      });
    }
    for (let i = 0; i < functionList.length; i++) {
        result += functionList[i]();
    }
    // Output in Ladybird is: 999999999.
    // Output should be: 012345678.
    console.log(result); 
</script>
```
I initially spotted this while working on #23544, looking at [this WPT subtest](https://github.com/web-platform-tests/wpt/blob/1efa678731e786c00bb5b60390166d98e7306175/dom/abort/resources/abort-signal-any-tests.js#L164-L184).

---

This was copied from: https://github.com/SerenityOS/serenity/issues/23552

**Comment by awesomekling** [comment]
This comment from @Lubrsi is important, as it describes the actual problem better:

------

This could potentially be us missing CreatePerIterationEnvironment: https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-createperiterationenvironment

For example, logging `result` in the function shows that result is captured by reference instead of by value:
```js
    let result = "";
    let functionList = [];
    for (let i = 0; i < 9; i++) {
      result += i;
      functionList.push(function() {
          console.log(result);
          return i;
      });
    }
    for (let i = 0; i < functionList.length; i++) {
        result += functionList[i]();
    }
    console.log(result); 
```
Chrome:
![image](https://github.com/SerenityOS/serenity/assets/25595356/cce2c391-847b-47ab-8b5d-9fdc1dbf11dd)
Ladybird:
![image](https://github.com/SerenityOS/serenity/assets/25595356/73285c13-14a0-4f77-ade9-bfbb2088ea1b)

**Comment by tcl3** [comment]
Fixed by dbc2f7e
